attachments/simple_engine/mesh_component.h

```diff
@@ -0,0 +1,461 @@
+ */
+struct Vertex {
+    glm::vec3 position;
+    glm::vec3 normal;
+    glm::vec2 texCoord;
+    glm::vec4 tangent;
+
+    bool operator==(const Vertex& other) const {
+        return position == other.position &&
+               normal == other.normal &&
+               texCoord == other.texCoord &&
+               tangent == other.tangent;
+    }
+
+    static vk::VertexInputBindingDescription getBindingDescription() {
+        vk::VertexInputBindingDescription bindingDescription(
+            0,                                  // binding
+            sizeof(Vertex),                     // stride
+            vk::VertexInputRate::eVertex        // inputRate
+        );
+        return bindingDescription;
+    }
+
+    static std::array<vk::VertexInputAttributeDescription, 4> getAttributeDescriptions() {
+        std::array<vk::VertexInputAttributeDescription, 4> attributeDescriptions = {
+            vk::VertexInputAttributeDescription{
+                .location = 0,
+                .binding = 0,
+                .format = vk::Format::eR32G32B32Sfloat,
+                .offset = offsetof(Vertex, position)
+            },
+            vk::VertexInputAttributeDescription{
+                .location = 1,
+                .binding = 0,
+                .format = vk::Format::eR32G32B32Sfloat,
+                .offset = offsetof(Vertex, normal)
+            },
+            vk::VertexInputAttributeDescription{
+                .location = 2,
+                .binding = 0,
+                .format = vk::Format::eR32G32Sfloat,
+                .offset = offsetof(Vertex, texCoord)
+            },
+            vk::VertexInputAttributeDescription{
+                .location = 3,
+                .binding = 0,
+                .format = vk::Format::eR32G32B32A32Sfloat,
+                .offset = offsetof(Vertex, tangent)
+            }
+        };
+        return attributeDescriptions;
+    }
+};
+
+/**
+ * @brief Component that handles the mesh data for rendering.
+ */
+class MeshComponent : public Component {
+private:
+    std::vector<Vertex> vertices;
+    std::vector<uint32_t> indices;
+
+    // Cached local-space AABB
+    glm::vec3 localAABBMin{0.0f};
+    glm::vec3 localAABBMax{0.0f};
+    bool localAABBValid = false;
+
+    // All PBR texture paths for this mesh
+    std::string texturePath;           // Primary texture path (baseColor) - kept for backward compatibility
+    std::string baseColorTexturePath;  // Base color (albedo) texture
+    std::string normalTexturePath;     // Normal map texture
+    std::string metallicRoughnessTexturePath;  // Metallic-roughness texture
+    std::string occlusionTexturePath;  // Ambient occlusion texture
+    std::string emissiveTexturePath;   // Emissive texture
+
+    // Instancing support
+    std::vector<InstanceData> instances;  // Instance data for instanced rendering
+    bool isInstanced = false;             // Flag to indicate if this mesh uses instancing
```

Comment:
Is this flag needed?
That is, can `instances` be non-empty, but `isInstances` is false?
`isInstances` being true with `instances` being empty is probably an error.